File: backend/app/schemas/documento.py
"""Schemas Pydantic para documentos anexados às ordens — US-015.

Importante: DocumentoResponse NÃO expõe o campo storage_path.
O acesso ao arquivo é feito exclusivamente via URL assinada (endpoint /download-url).
"""

import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field


class DocumentoResponse(BaseModel):
    """Representação pública de um documento anexado a uma ordem.

    US-015: storage_path NUNCA incluído — acesso somente via /download-url.
    """

    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    ordem_id: uuid.UUID
    uploaded_by: uuid.UUID
    nome_arquivo: str
    tipo_mime: str
    tamanho_bytes: int
    descricao: str | None
    hash_sha256: str
    assinado_govbr: bool
    versao: int
    created_at: datetime


class DocumentoListResponse(BaseModel):
    """Resposta de listagem de documentos de uma ordem."""

    documentos: list[DocumentoResponse]
    total: int = Field(description="Total de documentos na ordem")


class DownloadUrlResponse(BaseModel):
    """URL assinada para download seguro do arquivo.

    US-015: TTL configurável via SIGNED_URL_TTL_SECONDS (padrão: 900s / 15 min).
    """

    signed_url: str = Field(description="URL temporária para download do arquivo")
    expires_in: int = Field(description="Tempo de expiração da URL em segundos")

File: frontend/src/types/documento.ts
/**
 * Tipos de Documentos anexados às Ordens — US-015.
 *
 * Sincronizado com o back-end: app/schemas/documento.py e app/models/documento.py.
 */

// ---------------------------------------------------------------------------
// Entidades
// ---------------------------------------------------------------------------

/**
 * Metadados de um documento (GET /api/ordens/:id/documentos).
 *
 * US-015 RN: storage_path NUNCA é retornado pela API.
 * Acesso ao conteúdo somente via /download-url.
 */
export interface Documento {
  id: string
  ordem_id: string
  uploaded_by: string
  nome_arquivo: string
  tipo_mime: string
  tamanho_bytes: number
  descricao: string | null
  hash_sha256: string
  /** US-015: documento pode ser assinado digitalmente via gov.br. */
  assinado_govbr: boolean
  versao: number
  created_at: string
}

/** Resposta paginada de listagem (GET /api/ordens/:id/documentos). */
export interface DocumentoListResponse {
  documentos: Documento[]
  total: number
}

/** Resposta do endpoint de URL assinada (GET /api/documentos/:id/download-url). */
export interface DownloadUrlResponse {
  /** URL temporária para download direto. TTL: 900s (15 min). */
  signed_url: string
  /** Segundos restantes até a URL expirar. */
  expires_in: number
}

// ---------------------------------------------------------------------------
// Payloads de request
// ---------------------------------------------------------------------------

/**
 * FormData para POST /api/ordens/:id/documentos.
 *
 * Enviado como multipart/form-data (não JSON).
 */
export interface DocumentoUploadPayload {
  file: File
  descricao?: string
  /** true = documento já foi assinado via gov.br/assinatura. */
  assinado_govbr?: boolean
}
